lib/membrane_h264_plugin/parser.ex

```diff
@@ -241,7 +103,7 @@ defmodule Membrane.H264.Parser do
             state
             | mode: mode,
               nalu_splitter: NALuSplitter.new(input_stream_structure),
-              nalu_parser: NALuParser.new(input_stream_structure),
+              nalu_parser: NALuParser.new(state.encoding, input_stream_structure),
```

Comment:
we could resolve that in the specific parser implementation and pass a `NALuParser` module as an argument here